client/db/src/lib.rs

```diff
@@ -1974,6 +1978,14 @@ impl<Block: BlockT> sc_client_api::backend::Backend<Block> for Backend<Block> {
 		Ok(())
 	}
 
+	fn expose_db(&self) -> Option<Arc<dyn Database<DbHash>>> {
+		Some(self.db.clone())
+	}
+
+	fn expose_storage(&self) -> Option<Arc<dyn sp_state_machine::Storage<HashFor<Block>>>> {
+		Some(self.storage.clone())
+	}
```

Comment:
Do we really need these? Can we not just pass these values into the function where they are needed?

Reply:
Thanks for looking at it already :smile: I will refactor a bit before this is review ready.

The `sc_client::new_full_parts` from substrate wraps the DB and Storage and afterwards they are not accessible anymore.  
Not even to the Polkadot node.  
I could refactor `new_full_parts` to also return it, would be a bit cleaner than hacking the trait here.  

We have other parts where something like this is done, so I went for it in the first iteration; `Externalities.commit` for example.

Reply:
We could probably add the methods directly to the `Backend` struct?

Reply:
Oh sure :man_facepalming:, it returns the struct and not the backend trait. Fixed.